responsibleai/responsibleai/managers/counterfactual_manager.py

```diff
@@ -664,8 +666,10 @@ def _load(path, rai_insights):
 
         # Rehydrate model analysis data
         inst.__dict__[CounterfactualManager._MODEL] = rai_insights.model
-        inst.__dict__[CounterfactualManager._TRAIN] = rai_insights.train
-        inst.__dict__[CounterfactualManager._TEST] = rai_insights.test
+        inst.__dict__[CounterfactualManager._TRAIN] = \
+            rai_insights.train.drop(columns=rai_insights.metadata_columns)
```

Comment:
I'm not sure about what pandas does behind the scenes (or for that matter when Python does references or values) but.... we have this happening a lot, and it's going to be doing copies each time. Should we introduce `RAIInsights._X_test = RAIInsights.train.drop(columns=rai_insights.metadata_columns)` and just do it once at top level?

Reply:
This does do a copy and it's a valid concern. I can try using a view like this:
```
all_cols = df.columns
cols=list(set(all_cols) - set(metadata_columns))
df[cols]  # this is a view
```
However, I have to try whether that works at all. You can't modify views without getting the following:
```
<stdin>:1: SettingWithCopyWarning:
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
```
and I don't know whether we're trying to do something anywhere. Will report back.

Reply:
I changed every occurrences of drop to a view through `df[cols]` and it didn't complain so I infer from that that we're not changing the underlying DataFrame at any later point which is great. This should also help performance. Thanks for the suggestion!